packages/protocol/contracts/tokenvault/BridgedERC20.sol

```diff
@@ -70,22 +79,66 @@ contract BridgedERC20 is
         srcDecimals = _decimals;
     }
 
+    function startOutboundMigration(address to)
+        external
+        nonReentrant
+        whenNotPaused
+        onlyFromNamed("erc20_vault")
+    {
+        if (migratingTo != address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        if (to == address(0)) revert BRIDGED_TOKEN_INVALID_PARAMS();
+
+        migratingTo = to;
+        emit Migration(migratingFrom, migratingTo);
+    }
+
+    function startInboundMigration(address from)
+        external
+        nonReentrant
+        whenNotPaused
+        onlyFromNamed("erc20_vault")
+    {
+        if (migratingFrom != address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        if (from == address(0)) revert BRIDGED_TOKEN_INVALID_PARAMS();
+
+        migratingFrom = from;
+        emit Migration(migratingFrom, migratingTo);
+    }
+
+    function stopInboundMigration() external nonReentrant whenNotPaused onlyOwner {
+        if (migratingFrom == address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        migratingFrom = address(0);
+        emit Migration(migratingFrom, migratingTo);
+    }
+
     /// @notice Mints tokens to an account.
-    /// @dev Only an ERC20Vault can call this function.
     /// @param account The account to mint tokens to.
     /// @param amount The amount of tokens to mint.
-    function mint(address account, uint256 amount) public onlyFromNamed("erc20_vault") {
+    function mint(address account, uint256 amount) public nonReentrant whenNotPaused {
```

Comment:
So with this function what we basically say is:
- We can mint (into this token) if there is no incoming migration AND caller is the ERC20Vault
OR
- We can mint if there is an incoming migration (into this contract) AND the old contract is calling it.

But this is the "new" contract, no ? Shall not we allow incoming mints from the ERC20Vault, too ?

Reply:
updated the logics, please review again.